### PR TITLE
fix GPU other build

### LINKED ISSUE #12777: [ci] `gpu_other_build.sh` isn't running
See discussion in #12644 for context, this test was being silently skipped before and has broken in the meantime. We should fix it and re-enable it

cc @Mousius @areusch @gigiblender

### PR DESCRIPTION
**Pull request by alter-xp**
fixes #12777

@driazati @areusch

### CODE REVIEW

File: ci/jenkins/Build.groovy.j2 [hunk 1]
```diff
@@ -93,8 +93,9 @@ stage('Build') {
     {{ m.upload_artifacts(tag='gpu', filenames=tvm_multilib, folders=microtvm_template_projects) }}
 
     // compiler test
-    sh "${docker_run} --no-gpu ${ci_gpu} ./tests/scripts/task_config_build_gpu_other.sh build2"
-    make("${ci_gpu} --no-gpu", 'build2', '-j2')
+    sh "${docker_run} --no-gpu ${ci_gpu} ./tests/scripts/task_clean.sh build",
+    sh "${docker_run} --no-gpu ${ci_gpu} ./tests/scripts/task_config_build_gpu_other.sh build"
+    make("${ci_gpu} --no-gpu", 'build', '-j2')
     {{ m.upload_artifacts(tag='gpu2', filenames=tvm_multilib) }}
```
Comment (driazati) [inline ci/jenkins/Build.groovy.j2:99]:
```suggestion
```
these don't get used so they don't need to be uploaded

**Review by driazati — APPROVED** [review]
thanks for the fix! one small comment on another cleanup we can do

**Comment by driazati** [comment]
fyi @alter-xp we had to revert this in #13261 since it was breaking CI, can you re-open and fix the issues? I'll have to sync your changes to a branch in apache/tvm to properly test them in CI before merging (the `tvm-ci/branch` job)

**Comment by alter-xp** [comment]
@driazati Oops, I'm sorry for the problem caused by this PR.  let me fix it.

**Pull request merged**